backend/models/class.model.ts

```diff
@@ -81,4 +82,19 @@ const ClassSchema: Schema = new Schema(
   { timestamps: true },
 );
 
+/* eslint-disable func-names */
+ClassSchema.pre("findOneAndDelete", async function (next) {
+  try {
+    const doc = await this.findOne(this.getQuery()).clone();
+    if (doc) {
+      // Delete all test sessions associated with class
+      await MgTestSession.deleteMany({ class: doc.id });
+    }
+  } catch (error) {
+    return next(error as CallbackError | undefined);
+  }
+
+  return next();
+});
+
```

Comment:
does this change -> we delete all associated test sessions when we delete a class? if so, can we update the description & link the appropriate ticket to reflect this? 

this is different from deleting test sessions associated with a class when it is archived, right?

Reply:
oops this code has already been merge - branch didn't update